packages/peregrine/lib/targets/__tests__/peregrine-targets.spec.js

```diff
@@ -0,0 +1,82 @@
+const {
+    buildModuleWith,
+    mockTargetProvider
+} = require('@magento/pwa-buildpack/lib/testHelpers');
+const declare = require('../peregrine-declare');
+const intercept = require('../peregrine-intercept');
+
+test('declares a sync target talons and intercepts transformModules', () => {
+    const targets = mockTargetProvider(
+        '@magento/peregrine',
+        (_, dep) =>
+            ({
+                '@magento/pwa-buildpack': {
+                    specialFeatures: {
+                        tap: jest.fn()
+                    },
+                    transformModules: {
+                        tap: jest.fn()
+                    }
+                }
+            }[dep])
+    );
+    declare(targets);
+    expect(targets.own.talons.tap).toBeDefined();
+    const hook = jest.fn();
+    // no implementation testing in declare phase
+    targets.own.talons.tap('test', hook);
+    targets.own.talons.call('woah');
+    expect(hook).toHaveBeenCalledWith('woah');
+
+    intercept(targets);
+    const buildpackTargets = targets.of('@magento/pwa-buildpack');
+    expect(buildpackTargets.transformModules.tap).toHaveBeenCalled();
+});
+
+test('enables third parties to wrap talons', async () => {
+    jest.setTimeout(10000); // sorry, buildModuleWith is slow
+    const talonIntegratingDep = {
+        name: 'goose-app',
+        declare() {},
+        intercept(targets) {
+            targets.of('@magento/peregrine').talons.tap(talons => {
+                talons.ProductFullDetail.useProductFullDetail.wrapWith(
+                    'src/usePFDIntercept'
+                );
+                talons.App.useApp.wrapWith('src/useAppIntercept');
+                talons.App.useApp.wrapWith('src/swedish');
+            });
+        }
+    };
+    const built = await buildModuleWith('src/index.js', {
```

Comment:
Here is one of the test helpers for unit testing extensions. It does a full Webpack build and gives you a component runnable in Node, so you can test how your extension affected the built and generated code.